src/components/movie-card/movie-card.jsx

```diff
@@ -1,11 +1,28 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import PropTypes from 'prop-types';
 import {Link} from "react-router-dom";
 
+import VideoPlayer from "../video-player/video-player";
 import movieProp from '../../types/movie.prop';
 
-const MovieCard = ({film, handleMovieCardMouseOver, handleMovieCardMouseLeave}) => {
-  const {id, name, posterImage} = film;
+const MovieCard = ({film, isActive, handleMovieCardMouseOver, handleMovieCardMouseLeave}) => {
+  const {id, name, posterImage, videoLink} = film;
+  const [shouldVideoPlay, setShouldVideoPlay] = useState(false);
+  let timerId;
+
+  const onVideoPlay = () => setShouldVideoPlay(() => true);
```

Comment:
```suggestion
  const onVideoPlay = () => setShouldVideoPlay( true);
```
использование функции для обновления было бы оправдано, если бы требовался [доступ к предыдущему значению](https://reactjs.org/docs/hooks-reference.html#functional-updates)